### PR TITLE
Add visualization-server service to lightweight deployment

### PR DESCRIPTION
**Pull request by ajchili**
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/pipelines/1844)

### CODE REVIEW

File: backend/src/apiserver/server/visualization_server.go [hunk 1]
```diff
@@ -74,5 +74,5 @@ func (s *VisualizationServer) generateVisualizationFromRequest(request *go_clien
 }
 
 func NewVisualizationServer(resourceManager *resource.ResourceManager) *VisualizationServer {
-	return &VisualizationServer{resourceManager: resourceManager, serviceURL: "http://visualization-service.kubeflow"}
+	return &VisualizationServer{resourceManager: resourceManager, serviceURL: "http://ml-pipeline-visualization.kubeflow"}
```
Comment (IronPan) [inline backend/src/apiserver/server/visualization_server.go]: let's use const for host. and ".kubeflow" should be the namespace passed from the argument.

File: manifests/kustomize/base/kustomization.yaml [hunk 2]
```diff
@@ -28,3 +28,5 @@ images:
   newTag: 0.1.25
 - name: gcr.io/ml-pipeline/inverse-proxy-agent
   newTag: 0.1.25
+- name: gcr.io/ml-pipeline/visualization-server
```
Comment (IronPan) [inline manifests/kustomize/base/kustomization.yaml]: remove this till 0.1.26

File: manifests/kustomize/base/pipeline/ml-pipeline-visualization-deployment.yaml [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+        app: ml-pipeline-visualizationserver
+      spec:
+        containers:
+        - env:
+          - name: POD_NAMESPACE
+            valueFrom:
+              fieldRef:
+                fieldPath: metadata.namespace
+          # TODO: Push image to GCR and obtain tag
+          image: gcr.io/ml-pipeline/visualization-server:0.1.15
```
Comment (IronPan) [inline manifests/kustomize/base/pipeline/ml-pipeline-visualization-deployment.yaml]: it's ok to set it to 0.1.26 here actually

File: manifests/kustomize/base/pipeline/ml-pipeline-visualization-deployment.yaml [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+    matchLabels:
+      app: ml-pipeline-visualizationserver
+  template:
+    metadata:
+      labels:
+        app: ml-pipeline-visualizationserver
+      spec:
+        containers:
+        - env:
+          - name: POD_NAMESPACE
```
Comment (IronPan) [inline manifests/kustomize/base/pipeline/ml-pipeline-visualization-deployment.yaml]: is this environment var needed for the server?

**Comment by ajchili** [comment]
@IronPan I implemented your suggested changes but I am not sure if the manner in which I did it is the best way. If there is a better way to do them please let me know!

File: backend/src/apiserver/client_manager.go [hunk 5]
```diff
@@ -51,6 +51,9 @@ const (
 
 	podNamespace          = "POD_NAMESPACE"
 	initConnectionTimeout = "InitConnectionTimeout"
+
+	visualizationServiceHost = "ML_PIPELINE_VISUALIZATIONSERVER_SERVICE_HOST"
+	visualizationServicePort = "ML_PIPELINE_VISUALIZATIONSERVER_SERVICE_PORT_HTTP"
```
Comment (IronPan) [inline backend/src/apiserver/client_manager.go]: is _HTTP needed?

File: backend/src/apiserver/main.go [hunk 6]
```diff
@@ -83,7 +83,7 @@ func startRpcServer(resourceManager *resource.ResourceManager) {
 	api.RegisterRunServiceServer(s, server.NewRunServer(resourceManager))
 	api.RegisterJobServiceServer(s, server.NewJobServer(resourceManager))
 	api.RegisterReportServiceServer(s, server.NewReportServer(resourceManager))
-	api.RegisterVisualizationServiceServer(s, server.NewVisualizationServer(resourceManager))
+	api.RegisterVisualizationServiceServer(s, server.NewVisualizationServer(resourceManager, getStringConfig(visualizationServiceHost), getStringConfig(visualizationServicePort), getDurationConfig(initConnectionTimeout)))
```
Comment (IronPan) [inline backend/src/apiserver/main.go]: break into multiple lines

File: backend/src/apiserver/server/visualization_server.go [hunk 7]
```diff
@@ -4,18 +4,22 @@ import (
...
 	"strings"
+	"time"
 )
 
 type VisualizationServer struct {
-	resourceManager *resource.ResourceManager
-	serviceURL      string
+	resourceManager    *resource.ResourceManager
+	serviceURL         string
+	isServiceAvailable bool
```
Comment (IronPan) [inline backend/src/apiserver/server/visualization_server.go]: add a todo mentioning why this flag and this flag willl be removed once the service is stable.

File: backend/src/apiserver/client_manager.go (hunk 5 above)
Comment (ajchili) [reply backend/src/apiserver/client_manager.go]: Yes, see attached screenshot.
image

Comment (IronPan) [reply backend/src/apiserver/client_manager.go]: can you use one without http suffix? for consistency.

Comment (ajchili) [reply backend/src/apiserver/client_manager.go]: whoops yeah, I didn't realize that that it could be dropped.

**Pull request merged**